### ISSUE TITLE
Sensor to indicate the update status of 'image.doorbell_event_image'

### ISSUE THREAD

**Issue by xNUTx** [fuatakgun/eufy_security#989]
Hi @fuatakgun, remember https://github.com/fuatakgun/eufy_security/issues/929 ?

When the doorbell is rang, image.doorbell_event_image is updated. This process uses some time as the picture needs to be downloaded, decoded, encoded and saved. As I am trying to use the doorbell event image to display on a openhasp driven LCD screen through automation I ran into variable delays here and there.

It would however be nice to have some kind of (input_select) sensor to read out with the following values:
- 'outdated' = when not being updated for a set period of time, this should be the default value it resets to;
- 'updating' = after 'binary_sensor.doorbell_ringing' changes to 'on' this should be the value to show it is busy updating.
- 'updated' = once the whole retrieving, unpacking, repacking and saving cycle is done this should be the status, indicating the image is updated and is ready to be retrieved. After hitting 'updated' it should cycle back to 'outdated' after 5 seconds or so.

The alternative is to use the Automation delay functionality, but it does not know when a delay is experienced in performing the update task on the image. My experience is that it either needs a big delay to be more consistant (but slow in the process) or it becomes hit-and-miss when the timeouts are kept as short as possible.

If this feature is added to the integration, I would then be able to simplify the automation used to display the camera image on the openhasp display considerably and make it much more reliable compared to what it is now.

I hope you find my suggestion useful and will be able to build it in relatively easy.

Thanks for your time, effort and energy!

**Comment by fuatakgun** [comment]
Thanks for the idea, can you have an automation around ringing event last changed vs picture last changed?

In the meantime, i will check the idea

**Comment by xNUTx** [comment]
I'm not sure how to compare the pictures themselves in the automation, but it feels it may be very resource intensive.

You can however build automation using the ringing event, yes. The problem is that if downloading or unpacking the image takes longer then you can reasonably expect (for whatever reason) you will inevitably be stuck with a static time delay. Maybe you can loop something to have a check, but this also feels pretty resource intensive.

Or am I misinterpreting your question?

**Comment by fuatakgun** [comment]
I mean 

```
{{ states.image.frontdoor_event_image.last_changed }}
{{ states.binary_sensor.frontdoor_motion_detected.last_changed }}
```

**Comment by xNUTx** [comment]
Yes, but it triggers more often even when the timestamp itself does not change... 👎 

I'm still pretty green when it comes down to HA and it's automation, so I will keep trying for a while. For now, the details on why my **CURRENT** attempt is not usable:

`this:
  entity_id: automation.doorbell_notification
  state: 'on'
  attributes:
    id: '1697539776560'
    last_triggered: '2023-11-08T18:25:26.970962+00:00'
    mode: single
    current: 0
    friendly_name: Doorbell Notification
  last_changed: '2023-11-08T16:52:36.049057+00:00'
  last_updated: '2023-11-08T18:26:27.528241+00:00'
  context:
    id: 01HER2Z2J8EFG8RH02PMR6J4RT
    parent_id: null
    user_id: null
trigger:
  id: sensor
  idx: '0'
  alias: null
  platform: state
  entity_id: image.doorbell_event_image
  from_state:
    entity_id: image.doorbell_event_image
    state: '2023-11-08T16:53:02.840234+00:00'
    attributes:
      access_token: fa4523f186b23c508b0412210b06d9d60f5ef263578892bdcd9afd769df2bdd9
      entity_picture: >-
        /api/image_proxy/image.doorbell_event_image?token=fa4523f186b23c508b0412210b06d9d60f5ef263578892bdcd9afd769df2bdd9
      friendly_name: Doorbell Event Image
    last_changed: '2023-11-08T16:53:02.842026+00:00'
    last_updated: '2023-11-08T18:25:26.969028+00:00'
    context:
      id: 01HER2X7DSG10D5J1TMDZSXKET
      parent_id: null
      user_id: null
  to_state:
    entity_id: image.doorbell_event_image
    state: '2023-11-08T16:53:02.840234+00:00'
    attributes:
      access_token: d7b56f37371b1e9ed258e5d1605bd3181fc5ee7eddaf1560111666b5665e859c
      entity_picture: >-
        /api/image_proxy/image.doorbell_event_image?token=d7b56f37371b1e9ed258e5d1605bd3181fc5ee7eddaf1560111666b5665e859c
      friendly_name: Doorbell Event Image
    last_changed: '2023-11-08T16:53:02.842026+00:00'
    last_updated: '2023-11-08T18:30:26.970387+00:00'
    context:
      id: 01HER36CCTN46JXFBEMZ83MQ3K
      parent_id: null
      user_id: null
  for: null
  attribute: null
  description: state of image.doorbell_event_image`

![Screenshot 2023-11-08 at 19 33 18](https://github.com/fuatakgun/eufy_security/assets/6306755/891ab505-bedb-49d7-8470-1314d2944041)
![Screenshot 2023-11-08 at 19 32 47](https://github.com/fuatakgun/eufy_security/assets/6306755/aede2c16-8b03-49be-a0fe-6063c790fd53)
![Screenshot 2023-11-08 at 19 32 35](https://github.com/fuatakgun/eufy_security/assets/6306755/2853a3dc-6baf-47d3-a96e-ec4d7c5cb9bf)
![Screenshot 2023-11-08 at 19 31 48](https://github.com/fuatakgun/eufy_security/assets/6306755/aa496fe9-43ff-4dea-a301-415c1194a059)

I am trying to figure out how to do this... it's pretty confusing.

**Comment by xNUTx** [comment]
`trigger:
  - platform: state
    entity_id:
      - image.doorbell_event_image
    id: sensor
    attribute: last_changed`

This is the trigger for my latest attempt...

**Comment by fuatakgun** [comment]
Can you create a template sensor and set an automation reacting on it?

**Comment by xNUTx** [comment]
YES!

That worked 👍 

"{{ states.image.doorbell_event_image.state }}" in a template sensor helper... and now it triggers correctly on an update there... weird that the automation does trigger on last_update and not on last_changed... while this is updated when the 'state' is updated. Is that me or is that just weird?

Though... I would not mind having the original feature request implemented... these workarounds make automation not very 'simple'...

**Comment by fuatakgun** [comment]
You might not need a template sensor but rather a template trigger in an automation, can you try it too?

Whatever you put in the template sensor, you can put it directly on the automation template trigger.

**Comment by xNUTx** [comment]
So I add a trigger like this?

"
platform: template
   value_template: "{{ states.image.doorbell_event_image.state }}"
"

Or does it need any other logic?

**Comment by fuatakgun** [comment]
Yes, that should be it, it means one less entity (template helper)

**Comment by xNUTx** [comment]
Unfortunately that does not trigger the automation... so it seems to be missing something there... (maybe a comparison, old and new value)

**Comment by fuatakgun** [comment]
To be honest, i am falling behind my commitments here so i don't want to give a promise for this as there are many other issues around video streaming, i will focus on those first. Let's keep this open and see the reaction from community

**Comment by xNUTx** [comment]
@fuatakgun I have a workaround. Focus on bugs first. I understand my friend.

Thank you for your help so far 👍

**Comment by xNUTx** [comment]
FYI, I just discovered that there still is a delay you need to keep in mind when using the workaround. It seems to update the timestamp before it updates the actual picture it refers to... maybe that is a bug you can easily fix to reverse actions?

**Comment by fuatakgun** [comment]
Actually, I am updating the timestamp whenever picture changed event is sent to integration and converting the bytes into the picture.

Home assistant might be caching the previous one, but not controlled from integration.

**Comment by xNUTx** [comment]
Interesting... I will play around with it a bit more and try to find what fixes the problem and what does not. 

I will report my findings here once I have them. I will be going on holiday and will likely be doing this in January.
